chore: button support for light mode

deboer-tim opened the pull request:
### What does this PR do?

Adding Button variables for light mode and initial values. I had to make a few judgement calls on what the light mode should look like, and what variables/variable names to expose - e.g. secondary buttons I went with a single color value for both border and text.

### Screenshot / video of UI

Dark:

https://github.com/containers/podman-desktop/assets/19958075/25acc513-7b08-4fa2-96d9-4b4e4975c3bb

Light:

https://github.com/containers/podman-desktop/assets/19958075/d353c56d-c6b5-4ede-a0b0-966b1c76b2d6

### What issues does this PR fix or reference?

Fixes #7183.

### How to test this PR?

Probably easiest if you drop something like this in to see all buttons/disabled at once:

```
    <Button type="primary">Primary</Button>
    <Button type="secondary">Secondary</Button>
    <Button type="danger">Danger</Button>
    <Button type="tab">Tab</Button>
    <Button type="link">Link</Button>
    <Button type="primary" disabled>Primary</Button>
    <Button type="secondary" disabled>Secondary</Button>
    <Button type="danger" disabled>Danger</Button>
    <Button type="tab" disabled>Tab</Button>
    <Button type="link" disabled>Link</Button>
```

- [x] Tests are covering the bug fix or the new feature

ekidneyrh commented on packages/main/src/plugin/color-registry.ts:
```diff
@@ -808,4 +808,86 @@ export class ColorRegistry {
+
+    this.registerColor(`${button}primary-bg`, {
+      dark: colorPalette.purple[600],
+      light: colorPalette.purple[500],
```
light: colorPalette.purple[600]

axel7083 replied:
Hey! Github has a feature where you can make a `suggestion` :) might be useful :)

ekidneyrh replied:
oh cool I'll look into it! Thanks :)

axel7083 replied:
Here is a short demo if that can help :)

https://github.com/containers/podman-desktop/assets/42176370/e772f762-d549-46df-9837-706227d3920e

deboer-tim replied:
Original suggestion merged.

ekidneyrh commented on packages/main/src/plugin/color-registry.ts:
```diff
@@ -808,4 +808,86 @@ export class ColorRegistry {
+    });
+    this.registerColor(`${button}primary-hover-bg`, {
+      dark: colorPalette.purple[500],
+      light: colorPalette.purple[600],
```
light: colorPalette.purple[500]

deboer-tim replied:
Done.

ekidneyrh commented on packages/main/src/plugin/color-registry.ts:
```diff
@@ -808,4 +808,86 @@ export class ColorRegistry {
+    });
+    this.registerColor(`${button}secondary-hover`, {
+      dark: colorPalette.purple[500],
+      light: colorPalette.purple[800],
```
light: colorPalette.purple[500]

deboer-tim replied:
Done.

ekidneyrh commented on packages/main/src/plugin/color-registry.ts:
```diff
@@ -808,4 +808,86 @@ export class ColorRegistry {
+    });
+    this.registerColor(`${button}disabled`, {
+      dark: colorPalette.charcoal[50],
+      light: colorPalette.gray[700],
```
light: colorPalette.gray[900]

deboer-tim replied:
Done.

ekidneyrh commented on packages/main/src/plugin/color-registry.ts:
```diff
@@ -808,4 +808,86 @@ export class ColorRegistry {
+      light: colorPalette.gray[900],
+    });
+    this.registerColor(`${button}danger-border`, {
+      dark: colorPalette.red[600],
```
dark: colorPalette.red[500]

deboer-tim replied:
Done.

ekidneyrh commented on packages/main/src/plugin/color-registry.ts:
```diff
@@ -808,4 +808,86 @@ export class ColorRegistry {
+    });
+    this.registerColor(`${button}danger-border`, {
+      dark: colorPalette.red[600],
+      light: colorPalette.red[600],
```
light: colorPalette.red[700]

deboer-tim replied:
Done.

ekidneyrh commented on packages/main/src/plugin/color-registry.ts:
```diff
@@ -808,4 +808,86 @@ export class ColorRegistry {
+      light: colorPalette.red[600],
+    });
+    this.registerColor(`${button}danger-bg`, {
+      dark: colorPalette.charcoal[700],
```
I would set both of these to transparent

deboer-tim replied:
Done.

ekidneyrh commented on packages/main/src/plugin/color-registry.ts:
```diff
@@ -808,4 +808,86 @@ export class ColorRegistry {
+      light: colorPalette.gray[200],
+    });
+    this.registerColor(`${button}danger-text`, {
+      dark: colorPalette.white,
```
dark: colorPalette.red[500],
light: colorPalette.red[700]

deboer-tim replied:
Done.

ekidneyrh commented on packages/main/src/plugin/color-registry.ts:
```diff
@@ -808,4 +808,86 @@ export class ColorRegistry {
+      light: colorPalette.black,
+    });
+    this.registerColor(`${button}danger-hover-bg`, {
+      dark: colorPalette.charcoal[400],
```
dark: colorPalette.red[600],
light: colorPalette.red[600]

deboer-tim replied:
Done.

ekidneyrh commented on packages/main/src/plugin/color-registry.ts:
```diff
@@ -808,4 +808,86 @@ export class ColorRegistry {
+    this.registerColor(`${button}danger-hover-bg`, {
+      dark: colorPalette.charcoal[400],
+      light: colorPalette.gray[400],
+    });
```
I would add another value four button-danger-hover-text and set it to white, or point to button-text for the hover state

ekidneyrh commented on packages/main/src/plugin/color-registry.ts:
```diff
@@ -808,4 +808,86 @@ export class ColorRegistry {
+    });
+    this.registerColor(`${button}danger-border-disabled`, {
+      dark: colorPalette.gray[700],
+      light: colorPalette.gray[700],
```
dark: colorPalette.charcoal[050],
 light: colorPalette.gray[900],

I think having a standard disabled colour across the theme would be beneficial. There are a few diff values throughout, but that can be a separate ticket.

deboer-tim replied:
Done.

ekidneyrh commented on packages/main/src/plugin/color-registry.ts:
```diff
@@ -808,4 +808,86 @@ export class ColorRegistry {
+      light: colorPalette.gray[700],
+    });
+    this.registerColor(`${button}danger-bg-disabled`, {
+      dark: colorPalette.charcoal[800],
```
I would set these as transparent too

deboer-tim replied:
Done.

ekidneyrh reviewed (changes requested):
Left a few colour changes. I think we should have the primary and secondary buttons follow the same colour scheme, just have the difference be that one is filled and one is a ghost button. 

I mentioned that we should have a standard colour across the board for disabled. There is a few different values throughout pd, I'll open a separate ticket for that. 

For the danger buttons it looks better when the text matches the outline, the values just had to be made a bit darker for that. 

For the ghost buttons I would have the bg be transparent and then become filled on hover. 

Thanks :)

axel7083 commented on packages/main/src/plugin/color-registry.ts:
```diff
@@ -808,4 +808,86 @@ export class ColorRegistry {
+
+    this.registerColor(`${button}primary-bg`, {
+      dark: colorPalette.purple[600],
+      light: colorPalette.purple[500],
```
```suggestion
      light: colorPalette.purple[600],
```

deboer-tim commented on packages/main/src/plugin/color-registry.ts:
```diff
@@ -237,7 +237,7 @@ export class ColorRegistry {
     this.initDetails();
     this.initTab();
     this.initModal();
-    this.initLink();
```
Yup, had a few merge conflicts and missed this the second time. Fixed.

deboer-tim commented:
Had to rebase to pick up other changes, so I included all of the suggestions and updated the images at the top. Please review again.

Only two things that I see still need to be tweaked:
- The same colour was being used for enabled and disabled danger button text so now disabled buttons are red. I'm assuming that wasn't on purpose and I should define a new entry that should be dark: colorPalette.charcoal[050], light: colorPalette.gray[900]?
- The secondary hover doesn't have enough difference in light mode. Maybe we can darken this, or need a hover bg?

benoitf reviewed (approved):
the code is ok, for colors, I let @ekidneyrh decide

ekidneyrh commented:
>     * The same colour was being used for enabled and disabled danger button text so now disabled buttons are red. I'm assuming that wasn't on purpose and I should define a new entry that should be dark: colorPalette.charcoal[050], light: colorPalette.gray[900]?

Yah the disabled text would be those you mentioned above. For the secondary button it goes filled with disabled. Do we want to keep that and maybe have the danger button follow that style, or change it back to a ghost button and have it look like the danger button with the grey text? 

>     * The secondary hover doesn't have enough difference in light mode. Maybe we can darken this, or need a hover bg?

I mentioned above that for the ghost buttons I would suggest having the bg be transparent and then become filled on hover. Does it still not look like its being hovered on even then?

deboer-tim commented:
Rebased to pick up other changes in the color registry and added a commit to address the remaining issues:
- Fixed disabled button text.
- Misunderstood what you meant about ghost buttons before - added a hover effect to the secondary buttons now. The full '2' alpha on links was too bright and washed out the text and border so I went with '1' here, I think it's still enough?

Dropped new videos at the top so you can see the hover effects. @ekidneyrh please review again.

ekidneyrh commented:
I think I was explaining it poorly above. I meant to have the secondary button like what danger button is in the video, just with the purple colour instead. Like this:

![image](https://github.com/containers/podman-desktop/assets/87311656/ee518023-2bd9-42bc-843d-397c4902bc65)

Everything else looks great!

ekidneyrh reviewed (approved):
Looks great! Thanks :)

deboer-tim commented:
Thank you! Had to rebase again (no change) b/c too many PRs had gone into main.
